audit/hashstructure.go

```diff
@@ -27,76 +26,68 @@ func hashString(ctx context.Context, salter Salter, data string) (string, error)
 	return salt.GetIdentifiedHMAC(data), nil
 }
 
-// hashAuth returns a hashed copy of the logical.Auth input.
-func hashAuth(ctx context.Context, salter Salter, in *logical.Auth, HMACAccessor bool) (*logical.Auth, error) {
-	if in == nil {
-		return nil, nil
+// hashAuth uses the Salter to hash the supplied Auth (modifying it).
+// hmacAccessor is used to indicate whether the accessor should also be HMAC'd
+// when present.
```

Comment:
Godocs that explain all the arguments are _the best_.

<img src="https://media2.giphy.com/media/cPkUnVPxfmkMWzZfzx/giphy.gif"/>